Sprint-1/errors/3.js

```diff
@@ -1,9 +1,13 @@
-const cardNumber = 4533787178994213;
+const cardNumber = "4533787178994213";
 const last4Digits = cardNumber.slice(-4);
-
+console.log(`${last4Digits}`);
 // The last4Digits variable should store the last 4 digits of cardNumber
 // However, the code isn't working
 // Before running the code, make and explain a prediction about why the code won't work
 // Then run the code and see what error it gives.
 // Consider: Why does it give this error? Is this what I predicted? If not, what's different?
 // Then try updating the expression last4Digits is assigned to, in order to get the correct value
+
+
+//Answer : slice() function just works on arrays or strings, so we can define cardNumber as a string by using "" or use function
+//tostring() to convert a number to string
```

Comment:
Hi dear Sally. Thanks for reviewing my PR. I researched about it and found out that converting a number to a string can be useful, especially when dealing with data formats like card numbers that may need to retain leading zeros or be processed as text. However, there are potential issues to consider:
1-Data Validation: If I convert a number to a string,  I may lose the ability to perform mathematical operations directly. This could lead to errors if the string is used in calculations later.
2-Type Mismatch: If other parts of my system expect a number, passing a string instead can cause type errors, especially in strongly typed languages or when using strict type checking.
3-Comparison Issues: Comparing strings and numbers can yield unexpected results (e.g., "10" is less than "2" when compared as strings).
4-Consistency: Maintaining a consistent data type throughout my system is important for readability and maintainability. Mixing types can make the code harder to follow and debug.
In general, it's best to convert types when necessary, but to be cautious and aware of how those changes may affect other components of my system. :)